app/cdap/components/InlayDrawerWidget/index.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import { Menu } from '@material-ui/core';
+import ArrowDropDownIcon from '@material-ui/icons/ArrowDropDown';
+import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
+import {
+  Container,
+  Divider,
+  IconWrapper,
+  Label,
+  LeftContainer,
+  MenuButton,
+  StyledHeader,
+  StyledIconButton,
+  StyledMenuItem,
+} from 'components/InlayDrawerWidget/styles';
+import T from 'i18n-react';
+import React, { PropsWithChildren } from 'react';
+
+export interface IMenuItem {
+  clickHandler: () => void;
+  label: string;
+  value: string;
+}
+
+interface IRecipeStepWidgetProps {
+  actionsOptions: IMenuItem[];
+  headingText: string;
+  onClose: () => void;
+  position: 'left' | 'right';
+  disableActionsButton?: boolean;
+  showDivider?: boolean;
+}
+
+const getContainerComponent = (position: 'left' | 'right') => {
+  if (position === 'left') {
+    return LeftContainer;
+  }
+  return Container;
+};
+
+/**
+ *
+ * @param label - any space-separated string
+ * @returns - a hyphenated string
+ */
+export const getTestIdString = (label: string) =>
```

Comment:
this function is only used once?

Reply:
also with jsdoc you need a description

Reply:
> this function is only used once?

this function will be used in more places once we merge this code
I want to make this a util function
Once, we merge this code, we will reuse the same function across to get the strings for data-testid

Reply:
> also with jsdoc you need a description

added description to the jsdoc